core/error_macros.h

```diff
@@ -133,7 +133,7 @@ extern bool _err_error_exists;
 
 #define ERR_FAIL_INDEX(m_index, m_size)                                                                             \
 	do {                                                                                                            \
-		if (unlikely((m_index) < 0 || (m_index) >= (m_size))) {                                                     \
+		if (unlikely((m_index) < 0 || ((int)m_index) >= (m_size))) {                                                     \
```

Comment:
Seems good to me, but it should likely be done in a separate PR and also in `ERR_FAIL_INDEX_V`.
Existing casts to `int` could also be removed:
```
$ rg ERR_FAIL_INDEX.*\\\(\\\(int\\\)
scene/resources/animation.cpp
1917:   ERR_FAIL_INDEX((int)p_mode, 4);

scene/2d/camera_2d.cpp
452:    ERR_FAIL_INDEX((int)p_margin, 4);
459:    ERR_FAIL_INDEX_V((int)p_margin, 4, 0);
476:    ERR_FAIL_INDEX((int)p_margin, 4);
483:    ERR_FAIL_INDEX_V((int)p_margin, 4, 0);

scene/gui/texture_progress.cpp
62:     ERR_FAIL_INDEX((int)p_margin, 4);
69:     ERR_FAIL_INDEX_V((int)p_margin, 4, 0);

drivers/gles3/rasterizer_scene_gles3.cpp
4720:           ERR_FAIL_INDEX((int)shadow, shadow_atlas->quadrants[quadrant].shadows.size());

scene/gui/control.cpp
2251:   ERR_FAIL_INDEX((int)p_margin, 4);
2257:   ERR_FAIL_INDEX_V((int)p_margin, 4, NodePath());

scene/3d/audio_stream_player_3d.cpp
844:    ERR_FAIL_INDEX((int)p_mode, 2);

scene/resources/style_box.cpp
146:    ERR_FAIL_INDEX((int)p_margin, 4);
210:    ERR_FAIL_INDEX((int)p_expand_margin, 4);
233:    ERR_FAIL_INDEX_V((int)p_expand_margin, 4, 0);

drivers/gles2/rasterizer_scene_gles2.cpp
3006:           ERR_FAIL_INDEX((int)shadow, shadow_atlas->quadrants[quadrant].shadows.size());

scene/2d/collision_polygon_2d.cpp
218:    ERR_FAIL_INDEX((int)p_mode, 2);

scene/gui/label.cpp
513:    ERR_FAIL_INDEX((int)p_align, 4);
525:    ERR_FAIL_INDEX((int)p_align, 4);

scene/gui/line_edit.cpp
533:    ERR_FAIL_INDEX((int)p_align, 4);

scene/gui/nine_patch_rect.cpp
113:    ERR_FAIL_INDEX((int)p_margin, 4);
135:    ERR_FAIL_INDEX_V((int)p_margin, 4, 0);

core/os/input.cpp
45:     ERR_FAIL_INDEX((int)p_mode, 4);
```

Reply:
And same for `CRASH_BAD_INDEX` I guess.

Reply:
Alternatively you could just do the `(int)` cast in this PR, since it's already done in various other places.
We can then do and review the `error_macros.h` change in another PR, without blocking this one.

Reply:
And for ERR_FAIL_UNSIGNED_INDEX_V with cast to (unsigned int) I think...
OK, I'll do the PR then, unless somebody is working on it already.

Reply:
So in the current commit e9ac737fe, the cast is done at the macro call but with the intent of changing that later in another PR.

Reply:
Perfect, thanks!

Reply:
@NilsIrl I'm on it!